library/ssl_tls.c

```diff
@@ -3372,6 +3387,31 @@ const char *mbedtls_ssl_get_version(const mbedtls_ssl_context *ssl)
     }
 }
 
+#if defined(MBEDTLS_SSL_RECORD_SIZE_LIMIT)
+
+size_t mbedtls_ssl_get_output_record_size_limit(const mbedtls_ssl_context *ssl)
+{
+    const size_t max_len = MBEDTLS_SSL_OUT_CONTENT_LEN;
+    size_t record_size_limit = max_len;
+
+    if (ssl->session != NULL &&
+        ssl->session->record_size_limit >= MBEDTLS_SSL_RECORD_SIZE_LIMIT_MIN &&
```

Comment:
It's not clear to me under what circumstances we would put a value < `MBEDTLS_SSL_RECORD_SIZE_LIMIT_MIN` into the session. Would it make more sense to verify when populating `ssl->session->record_size_limit` and then be able to skip the check here?

Reply:
If we do not receive any record size limit extension then `ssl->session->record_size_limit` is equal to zero. Thus to me this check is to detect if the peer sent a record size limit or not.

Reply:
Ah - if this test had been for `> 0` then I might have thought that.

Would (in the future, not this PR) it make more sense to set this to the value to use for record size and use it unconditionally?

Reply:
Yes that would be an improvement.